lib/squadster/accounts/user.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Squadster.Accounts.User do
+  use Ecto.Schema
+
+  import Ecto.Changeset
+  require Logger
+  require Poison
+
+  alias Ueberauth.Auth
+  alias Squadster.Helpers.Dates
+
+  @vk_url "https://vk.com/"
+
+  schema "users" do
+    field :first_name, :string
+    field :last_name, :string
+    field :birth_date, :date
+    field :mobile_phone, :string
+    field :university, :string
+    field :faculty, :string
+    field :uid, :string
+    field :auth_token, :string
+    field :small_image_url, :string
+    field :image_url, :string
+    field :vk_url, :string
+    timestamps()
+  end
+
+  def changeset(params) do
+     changeset(%__MODULE__{}, params)
+  end
+
+  def changeset(%__MODULE__{} = struct, params \\ %{}) do
+    struct
+    |> cast(params, [:first_name, :last_name, :birth_date, :mobile_phone, :university, :faculty])
+    |> validate_required([:first_name, :last_name])
+    |> validate_format(:mobile_phone, ~r/^[-+()0-9]+$/)
+  end
+
+  def auth_changeset(params) do
+     auth_changeset(%__MODULE__{}, params)
+  end
+
+  def auth_changeset(%__MODULE__{} = struct, params \\ %{}) do
+    struct
+    |> cast(params, auth_fields())
+    |> validate_required([:uid, :first_name, :last_name])
+  end
+
+  def data_from_auth(%Auth{extra: %{raw_info: %{user: info}}, credentials: %{token: token}, info: %{phone: phone}} = auth) do
+    %{
+      first_name: info["first_name"],
+      last_name: info["last_name"],
+      birth_date: Dates.date_from_string(info["bdate"]),
+      mobile_phone: phone,
+      university: info["university_name"],
+      faculty: info["faculty_name"],
+      small_image_url: info["photo_100"],
+      image_url: info["photo_400"],
+      uid: uid_from_auth(auth),
+      vk_url: @vk_url <> (info["domain"] || ""),
+      auth_token: token
+    }
+  end
+
+  def uid_from_auth(%Auth{extra: %{raw_info: %{user: %{"id" => uid}}}}) do
+    Integer.to_string(uid)
+  end
+
+  defp auth_fields do
```

Comment:
maybe make sence to define it as module variables
cos it doesn't provide any logic, just return an array

Reply:
maybe
done